### ISSUE TITLE
recommend n-dash instead of m-dash?

### ISSUE THREAD

**Issue by danielnaber** [languagetool-org/languagetool#1551]
We suggest using m-dash here: e.g. in "It was - as far as I could tell - the only example of its kind."

A user says this is outdated, n-dash is now better and e.g. recommended by Oxford style guide.

@MikeUnwalla do you agree? If so, could you change the rule?

**Comment by MikeUnwalla** [comment]
From Oxford Manual of Style, 2002 edition, section 5.10.10 Dash:
OUP and most US publishers use the unspaced (non-touching) em rule as a parenthetical dash; other British publishers use the en rule with space either side.

I do not have a more recent copy of the Oxford Manual of Style.

The Wikipedia article for the rule (https://en.wikipedia.org/wiki/Dash#Em_dash) states, "Various style guides and national varieties of languages prescribe different guidance on dashes. Dashes have been cited as being treated differently in the US and the UK, with the former preferring the use of an em dash with no additional spacing and the latter preferring a spaced en dash"

So, I think that the rule is fine. It says, "Consider using..."

@TiagoSantos81 and @marcoagpinto worked on the rule. Possibly, they have comments.

**Comment by TiagoSantos81** [comment]
@MikeUnwalla I believe that the better solution is to add more em-dash and en-dash rules, conforming to the various styles. I haven't read anything about different country preferences, but we could start by moving the rule to the relevant country variant and adding an altered version to the other version. I created the rule for Portuguese, and @marcoagpinto suggested to port it to English, which seemed right.
According to https://www.thepunctuationguide.com/em-dash.html, and to my experience reading English books, the style is often seen in books of various sorts and newspapers. However, the spacing rules are different.
Also the en-dash is also very used by some editors. 
Summarizing, creating several variants and setting to `default='on'` only the most common version for each language variant is the way I suggest.

**Comment by MikeUnwalla** [comment]
@danielnaber, I agree with @TiagoSantos81 that "the better solution is to add more m-dash and n-dash rules, conforming to the various styles."

Ideally, for each 'authority' (MLA, Chicago, Oxford, and so on), we have a category or a rulegroup. But, to do that is a large amount of work. (And I don't know the style rules for dashes for the various authorities.)

At this stage, I suggest that the user deselects the rule if it is not useful.

**Comment by Tex2002ans** [comment]
@danielnaber @MikeUnwalla

> Ideally, for each 'authority' (MLA, Chicago, Oxford, and so on), we have a category or a rulegroup. But, to do that is a large amount of work. (And I don't know the style rules for dashes for the various authorities.)

There are effectively 2 "basic options".

You choose:

- EN DASH vs. EM DASH
  - British publications tend towards EN DASH + US publications tend towards EM DASH
  - "The food – which was delicious – reminded me of home." (British)
  - "The food—which was delicious—reminded me of home." (US)

Then if you choose EM DASH:

- certain publications prefer "unspaced" vs. "spaced" EM DASH:
  - "The food—which was delicious—reminded me of home." (unspaced)
    - This is CMOS's preference.
  - "The food — which was delicious — reminded me of home." (spaced)
     - This is the AP's preference, but it's typographically AWFUL. (See all the technical details in my posts below.)

That's the basics.

Here's [a post I wrote 8 months ago on Reddit](https://www.reddit.com/r/writing/comments/h7ja44/am_i_the_only_person_who_had_no_idea_there_were/fumpcq4/):

- - - -

[Wikipedia's article on "Dashes"](https://en.wikipedia.org/wiki/Dash#Em_dash) is a great reference.

**Dash/Hyphen Basics**

For the most part, it can be boiled down to this:

- — Em Dash
  - In Fiction when speakers get interrupted ("I did not shoot the—")
    For asides/parentheticals ("The food—which was delicious—reminded me of home.")
- – En Dash
  - Used between number ranges ("From 1950–1954")
- \- Hyphen
  - For everything else ("pre-hyphenated words")

That simple list should cover most uses.

If you want more details, the article above shows more examples of different Style Guides and nuances.

**Em Dash or Spaced En Dash?**

This is all personal preference.

Take the Em Dash example above:

> The food—which was delicious—reminded me of home.

this can also be written using an En Dash with spaces around it:

> The food – which was delicious – reminded me of home.

Different authors/publishers/Style Guides prefer one or the other. Whatever you do, just pick one and use it consistently throughout your work.

- - - -

I've also written about this extensively over the years. See:

- 2020: ["...Am I the only person who had no idea there were different types of dashes? I just use hyphens for everything..."](https://www.reddit.com/r/writing/comments/h7ja44/am_i_the_only_person_who_had_no_idea_there_were/fumpcq4/)
  - I laid down the basics of EM DASH vs. EN DASH vs. HYPHEN
  - Also listed common flaws in [Word/LibreOffice's autocorrect](https://www.reddit.com/r/writing/comments/h7ja44/am_i_the_only_person_who_had_no_idea_there_were/fumri33/)
- 2020: ["Punctuation Rules of Thumb?"](https://www.mobileread.com/forums/showthread.php?p=3945401#post3945401)
  - I explained hair/thin spaces in the "spaced em dash" (typographically correct, but much lower level support)
  - Also warned about MINUS SIGN. (A common error authors make, since "it *looks* like an dash". This error should be caught.)
- 2018: ["Punctuation is important too"](https://www.reddit.com/r/writing/comments/9q1jzi/punctuation_is_important_too/e8808lu/)
  - I referenced "University of Oxford Style Guide" + others.
  - (Note: This is NOT that "Oxford Manual of Style".)
- 2018: ["Sentence Structure Advice Needed"](https://www.reddit.com/r/writing/comments/9stytg/sentence_structure_advice_needed/e8rvh5a/)
  - I quoted CMOS's (17th edition, 6.87).
  - Fiction has an alternate style usage, em dashes used between dialogue.
  - See the fantastic article from [The Editor's blog: "When a Comma Isn't Enough"](https://theeditorsblog.net/2018/04/08/when-a-comma-isnt-enough/)

Foreign languages (like Spanish) get a bit more complicated, since they use a quotation dash.

- - - -

Anyway, Antidote 10 does a fantastic job catching these types of errors as well.

In their settings:

![Antidote 10 Dash Settings](https://user-images.githubusercontent.com/5774324/109337658-2b4c0680-7833-11eb-8b43-5c17f6c9cc80.png)

they have "Context":

- In front of each item in a list
- To set off a parenthetical expression
- To separate two clauses
- In a range
- To attribute a quote

You are then able to set "Type of dash":

- — em dash
- – en dash

and also "Space":

- none
- breaking
- non-breaking
- thin
- any

**Comment by MikeUnwalla** [comment]
> You choose:

I like the idea that the user chooses the style. It overcomes all the problems of making different rules for different style guides.
